src/rdf4cpp/rdf/datatypes/xsd/Double.cpp

```diff
@@ -15,6 +15,25 @@ std::string capabilities::Default<xsd_double>::to_string(cpp_type const &value)
     return util::to_chars(value);
 }
 
+template<>
+std::string capabilities::Default<xsd_double>::display(cpp_type const &value) noexcept {
+    if (value == 0) {
+        return std::signbit(value) ? "-0" : "0";
+    }
+
+    if (auto const abs = std::abs(value); abs >= 0.000001 && abs < 1000000) {
```

Comment:
Ah, [Casting to xs:string and xs:untypedAtomic](https://www.w3.org/TR/xpath-functions/#casting-to-string) it's handled here. Please state in CharConvExt  that the not the full behavior is covered.